Update hidden_explanation to include labels

NoRi2909 opened the pull request:
Currently the explanation for switching off "Visible" for entities is as follows:

![image](https://github.com/user-attachments/assets/96a4dd0c-b4f2-47b9-9950-b66fb90dd306)

The first part of the explanation is misleading as it says "… will not be shown on your dashboard …".

1. There are multiple dashboards today, so there is not a single one like this is implying.
2. You can still place any hidden entity on your dashboards.
3. So this refers to the auto-populated "Default dashboard" type, only.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue or discussion
  in the additional information section.
-->

The most important result of hiding an entity is that it is no longer included when its area, device or label is referenced.

This PR adds labels to the explanation and clarifies that hiding an entity only affects auto-populated dashboards.

## Type of change
<!--
  What type of change does your PR introduce to the Home Assistant frontend?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests


## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #22777 
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

<!--
  Thank you for contributing <3
-->

[docs-repository]: https://github.com/home-assistant/home-assistant.io

NoRi2909 committed:
Update hidden_explanation to include labels

The most important result of hiding an entity is that it no longer included when its area, device or label is referenced.

This commit adds labels to the explanation and drops the (obvious) part about the auto-populated dashboard that is misleading regarding all custom dashboards.

MindFreeze commented:
Not sure if entirely removing the dashboard mention is ok but clarifying it is not easy either

NoRi2909 commented:
Yes, a difficult decision. But as soon as you mention the word "dashboard" again you need to add quite some explanation that they do in fact not disappear from any user-created dashboard.

The auto-populated dashboard is for absolute novices, and they will either not touch that setting at all or when they turn off the "Visible" switch they will kind of expect them to be not included in the automatic dashboard.

NoRi2909 commented:
Digging further into the Dashboard issue I noticed that we have a duplicate use of "Default" regarding Dashboards:

The pre-populated dashboard is currently called "Default dashboard"

![Screenshot 2024-12-29 13 41 29](https://github.com/user-attachments/assets/8c378cf0-e1bd-4043-b8a9-1398e2fe120d)

and the one that shows up at the top of the Sidebar is also called "Default":

![Screenshot 2024-12-29 13 42 26](https://github.com/user-attachments/assets/2115e4b2-77a9-43f0-98fd-6a5d19276fa5)


Historically this was probably the same but it is no longer.  Therefore we should first rename the "Default dashboard" to "Pre-populated dashboard" or similar. And then it should be easier to keep the line:

"Hidden entities will not be included _in a pre-populated dashboard_ or when their area, device or label is referenced. …"

NoRi2909 commented:
I have created the new PR https://github.com/home-assistant/frontend/pull/23508 to rename the "Default dashboard" to "Automatic dashboard" so we could use this term here as suggested above.

NoRi2909 committed:
Add "auto-populated dashboards" back in

This re-adds the part of the dashboards but clarifies it using the term "auto-populated dashboards".

NoRi2909 commented:
The option to rename the current "Default dashboard" was postponed because several new dashboard types are planned for the near future. Thus my PR https://github.com/home-assistant/frontend/pull/23508 was closed.

As a result I've now re-added the part about the handling of hidden entities regarding dashboards, but clarified this part by adding "auto-populated". This makes it clear that user-defined dashboards are not affected which the current wording does imply.
